### PR TITLE
Issus/1598 automate release

### PR DESCRIPTION
**Pull request by hmottestad**
GitHub issue resolved: #1598   

Briefly describe the changes proposed in this PR:

* 
* 
*

### CODE REVIEW

**Comment by hmottestad** [comment]
what do you think of this kind of script @jeenbroekstra ?

File: release.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+read -p "Start the release process (y/n)?" choice
+case "${choice}" in
+  y|Y ) echo "";;
+  n|N ) exit;;
+  * ) echo "unknown response, exiting"; exit;;
+esac
+
+
+if  !  mvn -v | grep -q "Java version: 1.8."; then
+         echo "You need to use Java 8!";
```
Comment (jeenbroekstra) [inline release.sh]: I don't think we need to be this strict, exactly. The maven compiler is set to target 1.8, so even if you use a newer version of Java, it should still produce Java 8-compatible bytecode.

**Comment by hmottestad** [comment]
> Note, by the way, that our developer workflow documentation doesn't fully capture this. This is because previously, we got around the issue by just not recording the patch version in the snapshot - so we didn't have 3.0.1-SNAPSHOT and 3.0.2-SNAPSHOT, we just had 3.0-SNAPSHOT, so it doesn't need an update after release. I wanted to use more accurate snapshot versioning when we merged into a single repo, but perhaps we need to move back to that approach.

I think maybe we can handle this  creating a merge branch. After creating a release in master and incrementing the snapshot, we can create  a new branch where we set the version to the same as in develop and merge that branch into develop. Then merge master into develop. 

Need to test it, but in my head it sounds like it should work.

**Comment by jeenbroekstra** [comment]
> > Note, by the way, that our developer workflow documentation doesn't fully capture this. This is because previously, we got around the issue by just not recording the patch version in the snapshot - so we didn't have 3.0.1-SNAPSHOT and 3.0.2-SNAPSHOT, we just had 3.0-SNAPSHOT, so it doesn't need an update after release. I wanted to use more accurate snapshot versioning when we merged into a single repo, but perhaps we need to move back to that approach.
> 
> I think maybe we can handle this by creating a merge branch. After creating a release in master and incrementing the snapshot, we can create a new branch where we set the version to the same as in develop and merge that branch into develop. Then merge master directly into develop just to make git look pretty.
> 
> Need to test it, but in my head it sounds like it should work.

Another alternative approach is this: _before_ updating the pom version in master, merge master into develop (so that all meaningful changes are there). Then, update the version on master, and then merge with develop again using `git merge -s ours`.

File: release.sh (hunk 1 above)
Comment (jeenbroekstra) [reply release.sh]: As always this is not completely true as the build currently fails when using Java 11, of course. Keep it in, we can change it once we have build running on java 11 :)

**Comment by jeenbroekstra** [comment]
> Another alternative approach is this: _before_ updating the pom version in master, merge master into develop (so that all meaningful changes are there). Then, update the version on master, and then merge with develop again using `git merge -s ours`.

FWIW I just did this manually for the 3.0.1 release and it works perfectly.

**Comment by hmottestad** [comment]
One question about develop. I noticed that the <parent> tag points to 3.0.1-SNAPSHOT in develop even though the rest of the Pom references 3.1.0-SNAPSHOT. Is this the way it’s supposed to be?

**Comment by jeenbroekstra** [comment]
> One question about develop. I noticed that the tag points to 3.0.1-SNAPSHOT in develop even though the rest of the Pom references 3.1.0-SNAPSHOT. Is this the way it’s supposed to be?

I'm not sure what you mean. What tag are you referring to?

**Comment by hmottestad** [comment]
Huh. I was seeing it when I was creating the PR from the script. But now I can't reproduce it.

**Pull request merged**